### PR TITLE
Reuse lease in heartbeat store

### PR DESCRIPTION
**Pull request by cw9**
@xichen2020 @robskillington

### CODE REVIEW

File: services/heartbeat/etcd/store.go [hunk 1]
```diff
@@ -64,16 +96,16 @@ func (c *client) Get(service string) ([]string, error) {
...
 }
 
-func heartbeatKey(service, id string) string {
-	return fmt.Sprintf(keyFormat, servicePrefix(service), id)
+func heartbeatKey(service, instance string) string {
+	return fmt.Sprintf(keyFormat, servicePrefix(service), instance)
 }
 
-func instanceID(key, service string) string {
+func instanceFromKey(key, service string) string {
```
Comment (xichen2020) [inline services/heartbeat/etcd/store.go]: maybe just `instanceKey` to be consistent with `heartbeatKey` and `leaseKey`?

File: services/client/services.go [hunk 2]
```diff
@@ -143,12 +145,18 @@ func (c *client) Advertise(ad services.Advertisement) error {
...
+		errCounter := c.serviceTaggedScope(sid).Counter("heartbeat.error")
+
 		ticker := time.Tick(m.HeartbeatInterval())
 		for {
 			select {
 			case <-ticker:
 				if isHealthy(ad) {
-					hb.Heartbeat(serviceKey(ad.ServiceID()), ad.InstanceID(), m.LivenessInterval())
+					if err := hb.Heartbeat(serviceKey(sid), ad.InstanceID(), m.LivenessInterval()); err != nil {
+						c.logger.Infof("could not heartbeat service %s, %v", sid.String(), err)
```
Comment (xichen2020) [inline services/client/services.go]: Should this be an `Errorf`?

File: services/heartbeat/etcd/store.go (hunk 1 above)
Comment (cw9) [inline services/heartbeat/etcd/store.go]: this function is to parse instance id from the heartbeat key

File: services/client/services.go (hunk 2 above)
Comment (cw9) [inline services/client/services.go]: yeah good catch

File: services/heartbeat/etcd/store.go (hunk 1 above)
Comment (xichen2020) [inline services/heartbeat/etcd/store.go]: Fair enough

**Pull request merged**